source/MaterialXGenShader/HwShaderGenerator.cpp

```diff
@@ -122,9 +134,12 @@ ShaderPtr HwShaderGenerator::createShader(const string& name, ElementPtr element
     // Add the pixel stage output. This needs to be a color4 for rendering,
     // so copy name and variable from the graph output but set type to color4.
     // TODO: Improve this to support multiple outputs and other data types.
+    // TODO: If this is only outputing a fragment then we want to preserve the originl type
```

Comment:
I'm not sure this comment is needed here now? Sounds related to our fragment generation which is a separate thing.

Reply:
Yes, I should remove it.